synapse/storage/databases/main/roommember.py

```diff
@@ -363,37 +363,25 @@ async def get_local_current_membership_for_user_in_room(
         Returns:
             A tuple of (membership_type, event_id). Both will be None if a
                 room_id/user_id pair is not found.
-
         """
-        return await self.db_pool.runInteraction(
-            "get_local_current_membership_for_user_in_room",
-            self._get_local_current_membership_for_user_in_room_txn,
-            user_id,
-            room_id,
-        )
-
-    def _get_local_current_membership_for_user_in_room_txn(
-        self, txn, user_id: str, room_id: str
-    ) -> Tuple[Optional[str], Optional[str]]:
         # Paranoia check.
         if not self.hs.is_mine_id(user_id):
             raise Exception(
                 "Cannot call 'get_local_current_membership_for_user_in_room' on "
                 "non-local user %s" % (user_id,),
             )
 
-        sql = """
-            SELECT membership, event_id
-            FROM local_current_membership
-            WHERE
-                room_id = ? AND user_id = ?
-        """
+        results_dict = await self.db_pool.simple_select_one(
+            "local_current_membership",
+            {"room_id": room_id, "user_id": user_id},
+            ("membership", "event_id"),
+            allow_none=True,
+            desc="get_local_current_membership_for_user_in_room",
+        )
+        if not results_dict:
+            return None, None
 
-        txn.execute(sql, (room_id, user_id))
-        row = txn.fetchone()
-        if row:
-            return row[0], row[1]
-        return None, None
+        return results_dict.get("membership"), results_dict.get("event_id")
```

Comment:
Goodness this diff is hard to read. I've deleted the txn function and modified the original.

Reply:
The overall diff is OK though. 👍